### ISSUE TITLE
New Rule: If/else-expressions on same or new line

### ISSUE THREAD

**Issue by nickshanks** [eslint/eslint#17481]
### Rule details

Add/remove new line between expressions (not blocks) and if/else predicate

### Related ECMAScript feature

https://262.ecma-international.org/#prod-IfStatement

### What type of rule is this?

Enforces a formatting/stylistic preference

### Example code

Force new line mode, incorrect code:
```js
if (foo) return bar();
else baz = 'baz';
```

Force same line mode, incorrect code:
```js
if (foo)
  return bar();
else
  baz = 'baz';
```

The proposed rule would allow users to choose between enforcing one of the two styles above. Blocks would be unaffected, and rely upon the `brace-style` rule.

### Why should this rule be in the core instead of a plugin?

Because it applies universally to all javascript and typescript codebases.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

`if`, `else if`, and `else` are all to be treated equally. I recommend not setting a default, but requiring users to choose a style when enabling the rule.

**Comment by mdjermanovic** [comment]
Hi @nickshanks, thanks for the issue!

This rule already exists in the core: https://eslint.org/docs/latest/rules/nonblock-statement-body-position

* [Playground demo with `"below"`](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vbmJsb2NrLXN0YXRlbWVudC1ib2R5LXBvc2l0aW9uOiBbXCJlcnJvclwiLCBcImJlbG93XCJdICovXG5cbmZ1bmN0aW9uIGYoKSB7XG5cbiAgLy8gaW5jb3JyZWN0XG4gIGlmIChmb28pIHJldHVybiBiYXIoKTtcbiAgZWxzZSBiYXogPSAnYmF6JztcblxuICAvLyBjb3JyZWN0XG4gIGlmIChmb28pXG4gICAgcmV0dXJuIGJhcigpO1xuICBlbHNlXG4gICAgYmF6ID0gJ2Jheic7XG59XG5cbiIsIm9wdGlvbnMiOnsiZW52Ijp7ImVzNiI6dHJ1ZX0sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0In0sInJ1bGVzIjp7fX19)
* [Playground demo with `"beside"`](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vbmJsb2NrLXN0YXRlbWVudC1ib2R5LXBvc2l0aW9uOiBbXCJlcnJvclwiLCBcImJlc2lkZVwiXSAqL1xuXG5mdW5jdGlvbiBmKCkge1xuXG4gIC8vIGNvcnJlY3RcbiAgaWYgKGZvbykgcmV0dXJuIGJhcigpO1xuICBlbHNlIGJheiA9ICdiYXonO1xuXG4gIC8vIGluY29ycmVjdFxuICBpZiAoZm9vKVxuICAgIHJldHVybiBiYXIoKTtcbiAgZWxzZVxuICAgIGJheiA9ICdiYXonO1xufVxuXG4iLCJvcHRpb25zIjp7ImVudiI6eyJlczYiOnRydWV9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9LCJydWxlcyI6e319fQ==)

**Comment by nickshanks** [comment]
Ahh, thank you!
Perhaps it needs a more obious name?
